make I consistently provide identity behavior in type

Sacha0 opened the pull request:
Presently `I` is `UniformScaling{Int64}(1)`, which consistently provides identity behavior in value but inconsistently in type:
```julia
julia> typeof(Float32(1)I) # as expected
UniformScaling{Float32}

julia> typeof(Float32(1) + I) # as expected
Float32

julia> typeof(fill(Float32(1), 3, 3) + I) # as expected
Array{Float32,2}

julia> typeof(fill(Float32(1), 3, 3)I) # as expected
Array{Float32,2}

julia> typeof(Int32(1)I) # expect Int32, get Int64?
UniformScaling{Int64}

julia> typeof(Int32(1) + I) # expect Int32, get Int64?
Int64

julia> typeof(fill(Int32(1), 3, 3) + I) # expect eltype Int32, get eltype Int64?
Array{Int64,2}

julia> typeof(fill(Int32(1), 3, 3)I) # expect eltype Int32, get eltype Int64?
Array{Int64,2}

julia> typeof(fill(true, 3, 3)I) # expect eltype Bool, get eltype Int64?
Array{Int64,2}
```
This pull request makes `I` yield `UniformScaling{Bool}(true)` instead, which consistently provides identity behavior not merely in value but also in type. Ref. JuliaLang/LinearAlgebra.jl#454, and particularly the latter paragraphs in https://github.com/JuliaLang/LinearAlgebra.jl/issues/454, which this pull request partially addresses. Best!

Sacha0 committed:
Bind I to UniformScaling(true) rather than UniformScaling(1) for better promotion behavior.

fredrikekre reviewed (approved):
Technically breaking since `typeof(I)` is changing. Perhaps worth a `NEWS` entry?

Sacha0 commented:
> Technically breaking since typeof(I) is changing. Perhaps worth a NEWS entry?

Good call! Rebased out conflict and added a news entry. Thanks! :)

Given the approvals and 👍s/❤️s from relevant parties, I plan to merge this change this evening or tomorrow morning. Best!

StefanKarpinski commented on test/linalg/uniformscaling.jl:
```diff
@@ -51,7 +51,7 @@ end
 
 @testset "det and logdet" begin
-    @test det(I) === 1
+    @test det(I) === true
```
This is the only bit that gives me pause; I wonder if we should have `prod([true,true,true]) === 1` instead of `true`. We already have `sum([true,true,true]) === 3` so it would kind of make sense for the type of `prod` to match, even though `0` and `1` would be the only possible value.

Sacha0 replied:
Perhaps warrants a dedicated issue? :)

StefanKarpinski replied:
Done: https://github.com/JuliaLang/julia/issues/24425

Sacha0 commented:
Thanks all!
